src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -1100,10 +1105,20 @@ class FromClauseSegment(BaseSegment):
     parse_grammar = Sequence(
         "FROM",
         Delimited(
-            OneOf(
-                # check first for MLTableExpression, because of possible FunctionSegment in MainTableExpression
-                Ref("MLTableExpressionSegment"),
-                Ref("TableExpressionSegment"),
+            Sequence(
+                Indent,
+                OneOf(
+                    # check first for MLTableExpression, because of possible FunctionSegment in MainTableExpression
+                    Ref("MLTableExpressionSegment"),
+                    Ref("TableExpressionSegment"),
+                ),
+                Dedent.when(indented_joins=False),
+                AnyNumberOf(
+                    Ref("JoinClauseSegment"),
```

Comment:
I don't think that the join clause belongs here 😬. I actually think it belongs in the `TableExpressionSegment` per my previous PR: https://github.com/sqlfluff/sqlfluff/pull/791

Reply:
I hear you about making it part of table expression, _however_ we have a problem there where it becomes very recursive. Rather than replying here I'll add some more detail below.